cmd/gaia/cli_test/cli_test.go

```diff
@@ -139,6 +140,59 @@ func TestGaiaCLICreateValidator(t *testing.T) {
 	assert.Equal(t, "1/1", validator.PoolShares.Amount.String())
 }
 
+func TestGaiaCLISubmitProposal(t *testing.T) {
+
+	tests.ExecuteT(t, "gaiad unsafe_reset_all")
+	pass := "1234567890"
+	executeWrite(t, "gaiacli keys delete foo", pass)
+	executeWrite(t, "gaiacli keys delete bar", pass)
+	chainID := executeInit(t, "gaiad init -o --name=foo")
+	executeWrite(t, "gaiacli keys add bar", pass)
+
+	// get a free port, also setup some common flags
+	servAddr, port, err := server.FreeTCPAddr()
+	require.NoError(t, err)
+	flags := fmt.Sprintf("--node=%v --chain-id=%v", servAddr, chainID)
+
+	// start gaiad server
+	proc := tests.GoExecuteT(t, fmt.Sprintf("gaiad start --rpc.laddr=%v", servAddr))
+	defer proc.Stop(false)
+	tests.WaitForStart(port)
+
+	fooAddr, _ := executeGetAddrPK(t, "gaiacli keys show foo --output=json")
+	fooCech, err := sdk.Bech32ifyAcc(fooAddr)
+	require.NoError(t, err)
+
+	fooAcc := executeGetAccount(t, fmt.Sprintf("gaiacli account %v %v", fooCech, flags))
+	assert.Equal(t, int64(50), fooAcc.GetCoins().AmountOf("steak"))
+
+	executeWrite(t, fmt.Sprintf("gaiacli gov submitproposal %v --proposer=%v --deposit=5steak --type=Text --title=Test --description=test --name=foo", flags, fooCech), pass)
+	time.Sleep(time.Second * 2) // waiting for some blocks to pass
+
+	fooAcc = executeGetAccount(t, fmt.Sprintf("gaiacli account %v %v", fooCech, flags))
+	assert.Equal(t, int64(45), fooAcc.GetCoins().AmountOf("steak"))
+
+	proposal1 := executeGetProposal(t, fmt.Sprintf("gaiacli gov queryproposal --proposalID=1 --output=json %v", flags))
+	assert.Equal(t, int64(1), proposal1.ProposalID)
+	assert.Equal(t, "DepositPeriod", proposal1.Status)
+
+	executeWrite(t, fmt.Sprintf("gaiacli gov deposit %v --depositer=%v --deposit=10steak --proposalID=1 --name=foo", flags, fooCech), pass)
+	time.Sleep(time.Second * 2) // waiting for some blocks to pass
+
+	fooAcc = executeGetAccount(t, fmt.Sprintf("gaiacli account %v %v", fooCech, flags))
+	assert.Equal(t, int64(35), fooAcc.GetCoins().AmountOf("steak"))
+	proposal1 = executeGetProposal(t, fmt.Sprintf("gaiacli gov queryproposal --proposalID=1 --output=json %v", flags))
+	assert.Equal(t, int64(1), proposal1.ProposalID)
+	assert.Equal(t, "VotingPeriod", proposal1.Status)
+
+	executeWrite(t, fmt.Sprintf("gaiacli gov vote %v --proposalID=1 --voter=%v --option=Yes --name=foo", flags, fooCech), pass)
+	time.Sleep(time.Second * 2) // waiting for some blocks to pass
+
+	vote := executeGetVote(t, fmt.Sprintf("gaiacli gov queryvote  --proposalID=1 --voter=%v --output=json %v", fooCech, flags))
+	assert.Equal(t, int64(1), vote.ProposalID)
+	assert.Equal(t, "Yes", vote.Option)
```

Comment:
where is the test that requires bunch of blocks to pass before it will be successful? we should include that test (intending that it will fail) and write a note about exactly this.

Reply:
I realized that that's not actually needed.